app/v2/data.py

```diff
@@ -0,0 +1,80 @@
+import os
+from glob import glob
+
+import natsort
+import pandas as pd
+import xarray as xr
+
+from .constants import PATH_TO_FORECASTS, M3_TO_FT3
+
+__all__ = [
+    'get_forecast_dataset',
+    'get_return_periods_dataframe',
+    'find_available_dates',
+    'get_forecast_warnings_dataframe',
+    'latlon_to_reach',
+]
+
+
+def get_forecast_dataset(reach_id: int, date: str) -> xr.Dataset:
+    """
+    Opens the forecast dataset for a given date, selects the reach_id and Qout variable
+    """
+    if date == "latest":
+        date = find_available_dates()[-1]
+
+    forecast_file = os.path.join(PATH_TO_FORECASTS, f'Qout_{date}.zarr')
+
+    if not os.path.exists(forecast_file):
+        raise ValueError(f'Data not found for date {date}. Use YYYYMMDD format and the AvailableDates endpoint.')
+
+    try:
+        forecast_dataset = xr.open_zarr(forecast_file)
+    except Exception as e:
+        print(e)
+        raise ValueError('Error while reading data from the zarr files')
+    try:
+        return forecast_dataset.sel(rivid=reach_id).Qout
+    except Exception as e:
+        print(e)
+        raise ValueError(f'Unable to get data for reach_id {reach_id} in the forecast dataset')
+
+
+def get_return_periods_dataframe(reach_id: int, units: str) -> pd.DataFrame:
+    s3 = s3fs.S3FileSystem(anon=True, client_kwargs=dict(region_name=ODP_S3_BUCKET_REGION))
+    s3store = s3fs.S3Map(root=f'{ODP_S3_BUCKET_URI}/return-periods.zarr', s3=s3, check=False)
+    df = xr.open_zarr(s3store).sel(rivid=reach_id).to_dataframe()
+    if units == 'cfs':
+        for column in df:
+            df[column] *= M3_TO_FT3
+    return df
+
+
+def find_available_dates() -> list:
+    forecast_zarrs = glob(os.path.join(PATH_TO_FORECASTS, "Qout*.zarr"))
+    forecast_zarrs = natsort.natsorted(forecast_zarrs, reverse=True)
+    dates = [os.path.basename(d).replace('.zarr', '').split('_')[1] for d in forecast_zarrs]
+    return dates
+
+
+def get_forecast_warnings_dataframe(date) -> pd.DataFrame:
```

Comment:
This should probably raise a not implemented error for now

Reply:
Do we even copy these to the web server from the compute environment?